service/src/test/java/com/commercetools/util/HttpRequestUtilParallelTest.java

```diff
@@ -24,52 +26,108 @@
+                httpResponse -> assertThat(httpResponse.getStatusLine().getStatusCode()).isEqualTo(HttpStatus.SC_OK)));
     }
 
     @Test
     public void executePostRequest_shouldBeParalleled() throws Exception {
-        makeParallelRequests(nThreads, requests, () -> {
-            try {
-                final HttpResponse httpResponse = executePostRequest("http://httpbin.org/post", asList(
+        makeAndAssertParallelRequests(nThreads, requests, createRequestsSupplier(
+                () -> executePostRequest("http://httpbin.org/post", asList(
                         nameValue("xxx", "yyy"),
-                        nameValue("zzz", 11223344)));
-                assertThat(httpResponse.getStatusLine().getStatusCode()).isEqualTo(HttpStatus.SC_OK);
-                assertThat(responseToString(httpResponse)).contains("xxx", "yyy", "zzz", "11223344");
+                        nameValue("zzz", 11223344))),
+                // in post requests asserted status and body
+                httpResponse -> {
+                    assertThat(httpResponse.getStatusLine().getStatusCode()).isEqualTo(HttpStatus.SC_OK);
+                    assertThat(responseToString(httpResponse)).contains("xxx", "yyy", "zzz", "11223344");
+                }
+        ));
+    }
+
+    /**
+     * Function to create request/response supplier for GET/POST tests and assert the received response. The result of
+     * this function is used as a supplier for {@link #makeAndAssertParallelRequests(int, int, Supplier)}.
+     * <p>
+     * The function implementation takes care about {@code 408 Timeout} response from
+     * <a href="http://httpbin.org">http://httpbin.org</a>: if such response is received
+     * (when the service is overloaded) - the implementation makes a retry up to 3 times. This is important to avoid
+     * false failures of the tests when <a href="http://httpbin.org">http://httpbin.org</a> is overloaded.
+     *
+     * @param responseSupplier       supplier which when executed makes single blocking request and returns a response.
+     * @param responseAssertConsumer procedure to assert the response obtained from {@code responseSupplier}
+     * @return new supplier which when finished successfully - returns <b>1</b>, otherwise - <b>0</b>.
+     */
+    private Supplier<Integer> createRequestsSupplier(final ExceptionalSupplier<HttpResponse> responseSupplier,
+                                                     final ExceptionalConsumer<HttpResponse> responseAssertConsumer) {
+        return () -> {
+            try {
+                int statusCode;
+                int repeatedTimes = 0;
+                HttpResponse httpResponse;
+                do {
+                    if (repeatedTimes > 0) {
+                        LOGGER.warn("{} \"Request Timeout\" received from [{}]. Retry {} of {} ...",
+                                SC_REQUEST_TIMEOUT, "http://httpbin.org/", repeatedTimes, RETRY_TIMES);
```

Comment:
I see this string hardcoded 3 times in this file, do you think it makes sense to have a constant for it?

Reply:
are you talking about "http://httpbin.org/" ? or some other string?

Reply:
affc0a07

Reply:
> are you talking about "http://httpbin.org/" ?

Reply:
82455e0e441ff7d447ac13d4d0339766ac5f98fe